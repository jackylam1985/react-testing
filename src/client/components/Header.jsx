import * as React from 'react'
import { Link } from 'react-router-dom'

export default class Header extends React.Component<object, object> {
  shouldComponentUpdate () {
    return false
  }

  render () {
    return (
      <div className="site-header">
        <div className="branding-container">
          <a href="/" className="branding" />
        </div>
        <nav className="site-nav">
          <ul>
            <li>
              <Link to="/">Home</Link>
            </li>
            <li>
              <Link to="/node">Node</Link>
            </li>
          </ul>
        </nav>
      </div>
    )
  }
}
