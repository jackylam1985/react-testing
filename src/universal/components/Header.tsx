import * as React from 'react'
import { Link } from 'react-router-dom'

export default class Header extends React.Component<object, object> {
  render () {
    return (
      <div>
        <div>
          <a href="/" />
        </div>
        <nav>
          <ul>
            <li>
              <Link to="/">Home</Link>
            </li>
            <li>
              <Link to="/node">Node</Link>
            </li>
          </ul>
        </nav>
      </div>
    )
  }
}
