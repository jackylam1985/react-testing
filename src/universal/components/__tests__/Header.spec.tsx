// tslint:disable-next-line no-unused-variable
import * as React from 'react'
import { createRenderer } from 'react-test-renderer/shallow'
import Header from '../Header'

describe('components/Header', () => {
  it('renders', () => {
    const renderer = createRenderer()
    expect(renderer.render(
      <Header />,
    )).toMatchSnapshot()
  })
})
