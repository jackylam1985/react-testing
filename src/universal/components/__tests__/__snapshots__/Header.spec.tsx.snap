// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Header renders 1`] = `
<div>
  <div>
    <a
      href="/"
    />
  </div>
  <nav>
    <ul>
      <li>
        <Link
          replace={false}
          to="/"
        >
          Home
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/node"
        >
          Node
        </Link>
      </li>
    </ul>
  </nav>
</div>
`;
